### PR TITLE
Chore(api): change query builder named parameter

### PR DESCRIPTION
**Pull request by tleon**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Change named parameter in the module
| Type?             |  improvement 
| BC breaks?        | no
| Deprecations?    | no
| Fixed ticket?     | Fixes #35686
| Sponsor company   | PrestaShop SA
| How to test?      | CI 🟢

### CODE REVIEW

**Review by jolelievre — COMMENTED** [review]
Thanks @tleon 

All good, except for the services that should be moved from the core into here (see comment on the core PR)

File: config/admin/services.yml [hunk 1]
```diff
@@ -3,6 +3,25 @@ services:
     public: false
     autowire: true
 
-  PrestaShop\Module\APIResources\List\ModuleQueryBuilder:
+  prestashop.module.api.list.module_query_builder:
+    class: 'PrestaShop\Module\APIResources\List\ModuleQueryBuilder'
```
Comment (jolelievre) [inline config/admin/services.yml:7]:
```suggestion
  PrestaShop\Module\APIResources\List\ModuleQueryBuilder:
```

Comment (jolelievre) [reply config/admin/services.yml:7]: You can, and should, still rely on FQCN as the service name

File: config/admin/services.yml [hunk 2]
```diff
@@ -3,6 +3,25 @@ services:
...
+      - '@prestashop.core.api.query_builder.hook'
+      - '@prestashop.core.hook.dispatcher'
+      - '@prestashop.core.grid.query.doctrine_query_parser'
+      - 'hook'
+
+  prestashop.core.grid.data_factory.module:
+    class: '%prestashop.core.grid.data.factory.doctrine_grid_data_factory%'
+    public: true
+    arguments:
+      - '@prestashop.module.api.list.module_query_builder'
```
Comment (jolelievre) [inline config/admin/services.yml:24]:
```suggestion
      - '@PrestaShop\Module\APIResources\List\ModuleQueryBuilder'
```

**Review by jolelievre — APPROVED** [review]

**Review by jolelievre — DISMISSED** [review]

**Review by jolelievre — APPROVED** [review]

**Review by boherm — DISMISSED** [review]

**Review by jolelievre — APPROVED** [review]

**Pull request merged**